tests/ui/implicit_saturating_sub.rs

```diff
@@ -211,4 +226,39 @@ fn main() {
     } else {
         println!("side effect");
     }
+
+    // Extended tests
+    let mut m = Mock;
+    if m != 0 {
+        m -= 1;
+    }
+
+    let a = 12;
+    let mut b = 8;
+    let mut u_32 = 300;
+
+    if a < u8::MAX {
```

Comment:
The `implicit_sub_assign` lint should catch comparisons to zero, not MAX.

Reply:
Done. Comparing with `0`